src/main/java/net/minecraftforge/fml/common/discovery/ModDiscoverer.java

```diff
@@ -103,16 +100,14 @@ else if (modFile.isDirectory())
             }
             else
             {
-                Matcher matcher = zipJar.matcher(modFile.getName());
-
-                if (matcher.matches())
+                if (Files.getFileExtension(modFile.getName()).equals(".jar"))
```

Comment:
Instead of doing this you could pass FileFilterUtils.suffixFileFilter(".jar") at line 86